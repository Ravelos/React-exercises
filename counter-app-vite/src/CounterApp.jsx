import useState from 'react'
import PropTypes from 'prop-types'


export const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(value);                                                                     

    const handleAdd = () =>{
        // setCounter(counter + 1);
        // setCounter((c) => c + 1);
        setCounter(counter + 1)
    }

    const handleSubstract = () =>{
        setCounter(counter - 1)
    }

    const handleReset = () =>{
        setCounter(value)
    }

  return (
    <>
        <h1>Counter App</h1>
        <h2>{counter}</h2> 
        <button onClick={ handleAdd }>+1</button>     
        <button onClick={ handleSubstract }>-1</button>     
        <button onClick={ handleReset }>+Reset</button>     
    </>
  )
}

CounterApp.propTypes ={
    value: PropTypes.number
}
