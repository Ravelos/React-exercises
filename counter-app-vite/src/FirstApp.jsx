import PropTypes from 'prop-types'

export const FirstApp = ({title, subtitle}) => {
  return (
    <>
        <h1>{title}</h1>
        <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number
}

FirstApp.defaultProps = {
  // title:"There's no title",
  subtitle:'This is my subtitle'
}
