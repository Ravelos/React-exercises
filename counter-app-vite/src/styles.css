html, body{
    background-color: #36224a;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.5rem;
    padding: 70px;
}

button{
    padding: 5px;
    font-size: 1.3rem;
    margin-right: 5px;
}