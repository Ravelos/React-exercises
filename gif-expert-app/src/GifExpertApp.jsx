import { useState } from "react"
import AddCategory from "./components/AddCategory"
import { GifGrid } from "./components/GifGrid"

export const GifExpertApp = () => {

    const [ categories, setCategories ] = useState( [ 'One Punch', 'Dragon Ball' ] )
    
    const onAddCategory = (NewCategory) => {
        if( categories.includes(NewCategory)) return;
        setCategories([ NewCategory, ...categories]);  
    }

    return (
      <>
        <AddCategory onNewCategory={(event) => onAddCategory(event)} />

        <h1>GifExpertApp</h1>

        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </>
    );
}