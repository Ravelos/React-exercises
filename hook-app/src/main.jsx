import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import { HooksApp } from './HooksApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <HooksApp/>
  </React.StrictMode>,
)
