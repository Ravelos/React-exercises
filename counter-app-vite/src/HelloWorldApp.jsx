// export function App(){
//     return (
//         <h1>Hello Chanfle!!!</h1>
//     )
// }
import {FirstApp} from './FirstApp'


export const HelloWorldApp = () => {
  return (
    <>
        <h1>Hello world!!!</h1>
        <FirstApp />
    </>
  )
}