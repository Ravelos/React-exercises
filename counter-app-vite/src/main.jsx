import React from 'react'
import ReactDOM from 'react-dom/client';
// import { HelloWorldApp } from './HelloWorldApp'

import './styles.css'
import { CounterApp } from './CounterApp';
// import { FirstApp } from './FirstApp';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={20}/>
    </React.StrictMode>
)


