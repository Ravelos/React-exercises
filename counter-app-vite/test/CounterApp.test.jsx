import { fireEvent, render,screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Testing CounterApp component', () => {
    
    const value = 10;

    test('should match the snapshot', () => { 
        const {container } = render(<CounterApp value={value}/>);
        expect( container ).toMatchSnapshot();
    });

    test('should show the initial value of 100', () => {
        render(<CounterApp value={100}/>);
        expect( screen.getByText(100)).toBeTruthy();
    });
    
    test('should increment with +1 button', () => { 
        render(<CounterApp value={value}/>);
        fireEvent.click( screen.getByText('+1') );
        expect( screen.getByText('11')).toBeTruthy();
    });

    test('should increment with -1 button', () => { 
        render(<CounterApp value={value}/>);
        fireEvent.click( screen.getByText('-1') );
        expect( screen.getByText('9')).toBeTruthy();
    });

    test('should reset the value', () => {
        render(<CounterApp value={23}/>);
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        // fireEvent.click( screen.getByText('Reset') );

        fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }));
        expect( screen.getByText(23)).toBeTruthy();
    });

});