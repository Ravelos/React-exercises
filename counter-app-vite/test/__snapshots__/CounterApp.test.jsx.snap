// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CounterApp component should match the snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
